modules/gltf/editor/editor_scene_importer_gltf.cpp

```diff
@@ -49,18 +50,44 @@ void EditorSceneFormatImporterGLTF::get_extensions(List<String> *r_extensions) c
 Node *EditorSceneFormatImporterGLTF::import_scene(const String &p_path, uint32_t p_flags,
 		const HashMap<StringName, Variant> &p_options, int p_bake_fps,
 		List<String> *r_missing_deps, Error *r_err) {
-	Ref<GLTFDocument> doc;
-	doc.instantiate();
+	Ref<GLTFDocument> gltf;
+	gltf.instantiate();
+
+	Array document_extensions = p_options[SNAME("import_script/extensions")];
+	for (int i = 0; i < document_extensions.size(); i++) {
+		Ref<GLTFDocumentExtension> document_extension = document_extensions[i];
+		if (document_extension.is_valid()) {
+			gltf->get_extensions().append(document_extension);
+		} else if (r_err) {
+			*r_err = ERR_SCRIPT_FAILED;
+			ERR_PRINT("Invalid glTF document extension resource was specified.");
+		}
+	}
+
 	Ref<GLTFState> state;
 	state.instantiate();
-	Error err = doc->append_from_file(p_path, state, p_flags, p_bake_fps);
+	Error err = gltf->append_from_file(p_path, state, p_flags, p_bake_fps);
 	if (err != OK) {
 		if (r_err) {
 			*r_err = err;
 		}
 		return nullptr;
 	}
-	return doc->generate_scene(state, p_bake_fps);
+	return gltf->generate_scene(state, p_bake_fps);
+}
+
+Variant EditorSceneFormatImporterGLTF::get_option_visibility(const String &p_path, bool p_for_animation,
+		const String &p_option, const HashMap<StringName, Variant> &p_options) {
+	return true;
+}
+
+void EditorSceneFormatImporterGLTF::get_import_options(const String &p_path,
+		List<ResourceImporter::ImportOption> *r_options) {
+	if (p_path.get_extension().to_lower() != "gltf" && p_path.get_extension().to_lower() != "glb") {
+		return;
+	}
+
+	r_options->push_back(ResourceImporter::ImportOption(PropertyInfo(Variant::ARRAY, "import_script/extensions", PROPERTY_HINT_ARRAY_TYPE, vformat("%s/%s:%s", Variant::OBJECT, PROPERTY_HINT_RESOURCE_TYPE, "GLTFDocumentExtension")), Array()));
```

Comment:
I believe this can be done better. 

The current flaw with this is that we to type repeatedly per new gltf extended class.

```c++
r_options->push_back(ResourceImporter::ImportOption(PropertyInfo(Variant::ARRAY, "import_script/extensions", PROPERTY_HINT_ARRAY_TYPE, vformat("%s/%s:%s", Variant::OBJECT, PROPERTY_HINT_RESOURCE_TYPE, "GLTFDocumentExtension")), Array()));```

Any ideas how to remove this duplication?

Reply:
Is it possible to push from EditorSceneFormatImporterFBX and EditorSceneFormatImporterBlend to the superclass?

Reply:
> Any ideas how to remove this duplication?

I initially had it in `ResourceImporterScene::get_import_options` but I think not all scene formats will be using glTF behind the scenes (so having an array of `GLTFDocumentExtension` would not make sense for them). I haven't looked enough at the code to find out how it differs from `EditorSceneFormatImporter`, but I assume it would also be the same issue then?

I could make a superclass method like `add_document_extensions(p_document, p_extensions)` possibly, so it's a one-liner per importer?

> Is it possible to push this check ...

I assume you wanted to comment on the hardcoded `!= "gltf"` etc.? I can change the code to use the extensions list method yes.